poetry/console/commands/self/update.py

```diff
@@ -85,20 +87,23 @@ def update(self, release):
         prefix = sys.prefix
         base_prefix = getattr(sys, 'base_prefix', None)
         real_prefix = getattr(sys, 'real_prefix', None)
+        bin_dir = 'Scripts' if system() == 'Windows' else 'bin'
+        poetry_exe = 'poetry.exe' if system() == 'Windows' else 'poetry'
+        pip_exe = 'pip.exe' if system() == 'Windows' else 'pip'
```

Comment:
This might be easier to maintain and test as a small `Venv` wrapper class with these values as properties.

Reply:
Sorry, I don't quite follow. Are you having something like this in mind:
```
class Venv(object):
    @property
    def bin_dir(self):
        return 'Scripts' if system() == 'Windows' else 'bin'

…

        prefix_poetry = Path(prefix) / Venv().bin_dir / poetry_exe

```
What's the gain of it exactly?